### DISCUSSION TITLE
DNSMASQ and iPXE chainloading, ends up with "Operation not supported"

### DISCUSSION THREAD

**Discussion by DavidKabelac** [ipxe/ipxe#559] [Q&A] [answered]
Hello,
I'm trying to chainload DNSMASQ with iPXE to load into Windows setup, but whatever I try ends up with "Could not start download : Operation not supported (https://ipxe.org/3c092003)"..
I use DNSMASQ as TFTP, and also for DHCP proxy, and that will be the main problem I guess, just have no idea what to do with it. Also i use PXELINUX, all configurations are down below.

My DNSMASQ config: 

```
log-dhcp
dhcp-range=192.168.0.0,proxy
dhcp-boot=pxelinux.0                       

dhcp-match=set:iPXE,175
dhcp-boot=tag:iPXE,ipxe.lkrn 

pxe-prompt="Boot PXE",1

#PXEClient:Arch:00000
pxe-service=x86PC,'Boot legacy',pxelinux
#PXEClient:Arch:00007
pxe-service=BC_EFI,'Boot BC EFI,uefi/syslinux.efi
#PXEClient:Arch:00009
pxe-service=X86-64_EFI,'Boot x86-64 EFI',uefi/syslinux.efi

enable-tftp
tftp-root=/var/lib/tftpboot
```

DNSMASQ log: 

```
● dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
     Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-01-16 20:50:19 CET; 41min ago
    Process: 3270 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
    Process: 3271 ExecStart=/etc/init.d/dnsmasq systemd-exec (code=exited, status=0/SUCCESS)
    Process: 3280 ExecStartPost=/etc/init.d/dnsmasq systemd-start-resolvconf (code=exited, status=0/SUCCESS)
   Main PID: 3279 (dnsmasq)
      Tasks: 1 (limit: 8492)
     Memory: 536.0K
     CGroup: /system.slice/dnsmasq.service
             └─3279 /usr/sbin/dnsmasq -x /run/dnsmasq/dnsmasq.pid -u dnsmasq -I lo -7 /etc/dnsmasq.d,.dpkg-dist,.dpkg-old,.dpkg-new --local-service --trust-anchor=.,20326,8,2,e06d44b80b8f1d39a95c0b0d7c6>

led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 sent size: 42 option: 43 vendor-encap  06:01:03:08:07:80:00:01:c0:a8:00:6d:09:10...
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 available DHCP subnet: 192.168.0.0/255.255.255.0
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 vendor class: PXEClient:Arch:00000:UNDI:002001
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 user class: iPXE
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 available DHCP subnet: 192.168.0.0/255.255.255.0
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 vendor class: PXEClient:Arch:00000:UNDI:002001
led 16 21:31:01 pxe-System-Product-Name dnsmasq-dhcp[3279]: 117617450 user class: iPXE
led 16 21:31:24 pxe-System-Product-Name dnsmasq-dhcp[3279]: 2761353739 available DHCP subnet: 192.168.0.0/255.255.255.0
led 16 21:31:24 pxe-System-Product-Name dnsmasq-dhcp[3279]: 2761353739 vendor class: MSFT 5.0
led 16 21:31:24 pxe-System-Product-Name dnsmasq-dhcp[3279]: 2761353739 client provides name: PC
```
PXELINUX config:
```
default vesamenu.c32

prompt 0

menu title PXE Boot
label Windows
        menu label Windows 10 21H2 64-bit
        text help
Instalace Windows 21H2 ze site.
        endtext

        kernel ipxe/ipxe.lkrn
        append dhcp && chain ipxe/Win10H2.ipxe
```
Thank you for any useful tip.
David

**Comment by NiKiZe** [comment, marked as answer]
So you are able to start iPXE from PXELINUX?
But after this it fails?

On possibility is that you have `dhcp-boot=tag:iPXE,ipxe.lkrn`
Which sends ipxe.lkrn as filename, not that you don't have any protocol here, and not sure how dnsmasq does with this.
This is already covered by the documentation at https://ipxe.org/3c092003

When iPXE starts, do press Ctrl+B and try to do dhcp followed by config to show the actual configuration in regards to filename and next-server.
Using tcpdump or similar to catch the actual DHCP packets might also be a good idea.

One thing to consider tho is why you are using PXELINUX in the first place?
You can find a working dnsmasq proxy dhcp config at https://ipxe.org/appnote/proxydhcp



## Remember to reply to this answer and avoid creating a new answer unless you actually find a better solution

**Reply by DavidKabelac** [reply]
You are my hero! https://ipxe.org/appnote/proxydhcp solved the issue, and now I can boot Windows setup successfully.

I use pxelinux for custom menu.

Thanks a lot!

**Reply by mcb30** [reply]
@DavidKabelac You may be able to simplify your setup by using iPXE's native menu capability: see https://ipxe.org/cmd/choose and https://ipxe.org/cmd/item
